src/main/java/ServiceLayer/UserService.java

```diff
@@ -61,4 +64,22 @@ public Response register(String userName, String password, String email) {
         }
         return response;
     }
+
+    public Response purchaseCart(String token, List<Integer> busketsToBuy, PaymentMethod paymentMethod, SupplyMethod shippingMethod) {
+        Response response = new Response();
+        try {
+            String username = tokenService.getUsernameFromToken(token);
+            if (userController.isUserNameExists(username)) {
+                userController.getUserByUsername(username).purchaseCart(busketsToBuy, paymentMethod, shippingMethod);
+                logger.info("User purchase cart successfully: " + username);
+                response.setReturnValue("Purchase cart Succeed");
```

Comment:
you want to return the cart no? or the order?

Reply:
It needs to return order.